### PR TITLE
Fixes #83: test case for text-parser

### LINKED ISSUE #83: Test for issue #75
A simple test to check if the html can be converted to text plain once #75  is merge

Test are written in [Jest](https://jestjs.io/) 
Command to run test: **npm test**

Test is saved in **test** repo

### PR DESCRIPTION
**Pull request by giatuongtran9**
Adding test case for parsing html to text, reference to issue #75 

@PavanKKamra, apparently JSOM does not support innetText, based on this [Issue](https://github.com/jsdom/jsdom/issues/1245). I have tested it and innerHTML and textContent works fine but innerText will return 'undefined'

### CODE REVIEW

**Comment by giatuongtran9** [comment]
Hi @humphd , I have change the test to return a Promise by async await.

I have noticed that **dialogue.js** has some errors whenever I run the test 'npm run test'
Screen Shot 2019-11-08 at 7 35 31 PM

**Comment by humphd** [comment]
@giatuongtran9 there is a PR to fix that in #106 if you want to review.

**Comment by humphd** [comment]
@giatuongtran9 it looks like you've merged and pulled in a bunch of unrelated changes.  Also, there are merge conflict markers in `test/text-parser.test.js`.

NOTE: you want to rebase but not merge with master.

**Comment by giatuongtran9** [comment]
@humphd I made some mistakes at the beginning and try to rebase to master but somehow pulled all the new files. The test fail because we don't have the **text-parser.js** yet

**Comment by humphd** [comment]
OK, we'll have to fix this git history.  I can help you when we see each other at the start of the week, or you can try and do it yourself between now and then.

As far as the failure, tell Jest to `skip` these tests until we have an implementation: https://jestjs.io/docs/en/api.html#testskipname-fn

**Comment by giatuongtran9** [comment]
I will try to fix this but I think I will need your help. skip cannot be used here because the test need the run method from text-parser.js so we can comment out the test or do this for now:
```
expect('Hello World').toBe('Hello Word');
```

**Comment by humphd** [comment]
OK @giatuongtran9, how about this.  Since @PavanKKamra doesn't seem to have started his work yet, why don't you add `text-parser.js` and have it export a function at least.  In the code, you can leave a comment to the Issue that will implement it in #75. Then @PavanKKamra can work from there.

@PavanKKamra how is your work coming?  It sounds like we're just about ready for your code here...

**Comment by giatuongtran9** [comment]
@humphd  I have added simple function for text-parser and @PavanKKamra can continue from there

File: src/text-parser.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+const { JSDOM } = jsdom;
+
+
+// Simple function to convert html to text
+// Issue #75
+exports.run = async function (html) {
+  const dom = new JSDOM(html);
+  const result = dom.window.document.querySelector('body').textContent;
+
+  return result;
```
Comment (humphd) [inline src/text-parser.js]: `return Promise.resolve(result);` so it gives back a `Promise`.

File: src/text-parser.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const jsdom = require('jsdom');
+
+const { JSDOM } = jsdom;
+
+
```
Comment (humphd) [inline src/text-parser.js]: Remove this extra blank line, please.

File: test/text-parser.test.js [hunk 3]
```diff
@@ -4,16 +4,13 @@ const textParser = require('../src/text-parser');
 /**
  * textParser.run() will convert the html to text plain
  * Function should only extract what in the body
- * Fuction should only extract what in the body
  */
 test('Testing text-parser', async () => {
```
Comment (humphd) [inline test/text-parser.test.js]: Do you not need to `skip` this one too?

File: src/text-parser.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+// Simple function to convert html to text
+// Issue #75
```
Comment (humphd) [inline src/text-parser.js]: Replace this with a full GitHub URL to issue 75, please.

File: test/text-parser.test.js [hunk 5]
```diff
@@ -10,7 +10,8 @@ test('Testing text-parser', async () => {
   expect(result).toBe('Hello World');
 });
 
-test.skip('Testing text-parser with new line', async () => {
-  const result = await textParser.run('<!DOCTYPE html><html><head><title>OSD600</title></head><body style = "text-align:center;"><h1>Seneca</h1><div style="white-space: pre">OpenSource Telescope</div></body></html>');
+// Issue: Don't work with new line
```
Comment (humphd) [inline test/text-parser.test.js]: What does this mean?

Comment (giatuongtran9) [reply test/text-parser.test.js]: my function in text-parser.js won't convert the new line and I don't know why. I will fix it later

Comment (humphd) [reply test/text-parser.test.js]: I suspect it's due to limitations in JSDOM, which doesn't have any layout engine, and things like newlines depend on having one.  I left some notes in #75 on how we might overcome this later.

**Comment by humphd** [comment]
@giatuongtran9 you need to regenerate `package-lock.json` like so:

```
npm install
git add package-lock.json
```

**Comment by giatuongtran9** [comment]
@humphd Thank you and @cindyledev please review it again

**Comment by giatuongtran9** [comment]
@cindyledev please review again, if you left it here, other stuff will get merged and this will forever get conflict and never get merge
@humphd sorry for asking to review it again so many time

**Pull request merged**